dll/appcompat/apphelp/apphelp.c

```diff
@@ -240,3 +251,229 @@ ApphelpCheckRunAppEx(
     return TRUE;
 }
 
+
+/**
+ * @name SdbRegisterDatabaseEx
+ * Register an application compatibility database
+ *
+ * @param pszDatabasePath   The database. Required
+ * @param dwDatabaseType    The database type. SDB_DATABASE_*
+ * @param pTimeStamp        The timestamp. When this argument is not provided, the system time is used.
+ * @return                  TRUE on success, or FALSE on failure.
+ */
+BOOL WINAPI SdbRegisterDatabaseEx(
+    _In_ LPCWSTR pszDatabasePath,
+    _In_ DWORD dwDatabaseType,
+    _In_opt_ PULONGLONG pTimeStamp)
+{
+    PDB pdb;
+    DB_INFORMATION Information;
+    WCHAR GuidBuffer[50];
+    UNICODE_STRING KeyName;
+    ACCESS_MASK KeyAccess;
+    OBJECT_ATTRIBUTES ObjectKey = RTL_INIT_OBJECT_ATTRIBUTES(&KeyName, OBJ_CASE_INSENSITIVE);
+    NTSTATUS Status;
+    HANDLE InstalledSDBKey;
+
+    pdb = SdbOpenDatabase(pszDatabasePath, DOS_PATH);
+    if (!pdb)
+    {
+        SHIM_ERR("Unable to open DB %S\n", pszDatabasePath);
+        return FALSE;
+    }
+
+    if (!SdbGetDatabaseInformation(pdb, &Information) ||
+        !(Information.dwFlags & DB_INFO_FLAGS_VALID_GUID))
+    {
+        SHIM_ERR("Unable to retrieve DB info\n");
+        SdbCloseDatabase(pdb);
+        return FALSE;
+    }
+
+    if (!SdbGUIDToString(&Information.Id, GuidBuffer, RTL_NUMBER_OF(GuidBuffer)))
+    {
+        SHIM_ERR("Unable to Convert GUID to string\n");
+        SdbFreeDatabaseInformation(&Information);
+        SdbCloseDatabase(pdb);
+        return FALSE;
+    }
+
+    RtlInitUnicodeString(&KeyName, INSTALLEDSDB_KEYNAME);
+    KeyAccess = Wow64QueryFlag() | KEY_WRITE | KEY_QUERY_VALUE | KEY_ENUMERATE_SUB_KEYS;
```

Comment:
The Wow64QueryFlag function seems pointless here -- the NT API allows but simply ignores those flags.

Reply:
Guess I missed that when converting my code from advapi to ntdll functions :(
This isnt the only place I did this, will fix that later.